### ISSUE TITLE
Are large data transfers supported?

### ISSUE THREAD

**Issue by pwen090** [aspnet/SignalR#782]
In the previous (or current I guess) implementation of SignalR the hub framework was not recommend for doing any sort of large data transfers. Instead you would signal a client/server and then they would call a web api/REST to do anything that was related to large transfers of data.

I am wondering for this upcoming signalr asp.net core version what large data transfers would look like? Is it still not recommended and we should continue to just signal and call some web api/REST endpoint out of band?

I did not see any existing Issue related to this topic so hopefully I have not missed an existing answers. thank you

**Comment by davidfowl** [comment]
The limitations that were there before aren't the same. Though it would still be more efficient in some cases to make a REST call and download the data, the new SignalR is much closer to a raw websocket and supports binary protocols. 

We still haven't tested streaming since today we buffer the entire payload before calling your method but that's just a point in time thing. We'd need to carefully design this so that streaming is efficient. You wouldn't want to just send a large `byte[]` as part of an invocation as it won't scale.

**Comment by PRIMETSS** [comment]
Is there anywhere to go for more info on this sort of work load? (ASPNET .NET CORE SignalR)
I'm trying to use SignalR to broadcast datalogger acquisition data out to 3-5 clients. which runs at about 10-20 lines (eg "101*HV: 12.3v") a second and so far from logging server side events it seems to keep up, but JS side I'm getting browser slowdowns and eventiall lock ups.
I was using API calls, but stripped that out and was under the impression SignalR Core had improved its performance dramatically?

Have I chosen the wrong tech?

**Comment by davidfowl** [comment]
Even if SignalR's performance has improved dramatically it doesn't mean you don't need to worry about optimizing your own code for specific environments. If you overwhelm the browser with data, it'll still lock up. You still need to do all of your due diligence to make sure your own code is behaving properly.

That said, have you done any profiling?

**Comment by PRIMETSS** [comment]
Thanks for feedback David, no not yet, as Ive only just noticed its happening under load, as have been concentrating on removing the API side and using SignalR for both command-control and reading data back from the loggers. I need to find out whats causing the browser to literally lock up (takes about 30 secs for it to stop reporting site/page un-responsive, and then continue (after stopping the Hub from transmitting)
What I'm suspecting is that because the server is broadcasting data as it comes down from the logger (TCP socket stream connection (Telnet basically)) [about a 100 lines every 4-7 second intervals] the browser is still processing the last batch as the new one comes in, and slowly over time it gets slower & slower till it locks up. 
But I would not have thought that 100 records every 4 seconds would be too much...
So as you say, their might be another issue in my implementation
.
I'll look into doing some profiling and see if I can narrow down where the bottle neck is...

**Comment by PRIMETSS** [comment]
David, Just some feedback on my comment above.
Your absolutely corrrect!
I had difficulty doing any useful profiling, as the problem I had brought every browser to a grinding halt, which made fault finding it a challenge.
After spending a few days on it , and after upgrading the 5-6 project solution all over to Core 2.1 to get the latest SignalR bits (This project is for Internal R&D, so Im happy with where Core 2.1 I decided to move forward earlier than expected) [which moving to 2.1 was relatively easy with no or little breaking changes] {Also the Docs team has done a good job on the docs site, being able to select docs based on the version dropdown makes navigating all the mix of info far less painful!}
TLDR; To cut it short, it turned out to be a bit of UI I put in awhile back, to actually get a visual on the performance, and didnt notice it back then as an issue.
So it was a silly issue, self caused!
SignalrR is now performing well, and able to broadcast live data acquisition data to many listening clients with pleasing results

Thanks for your time, cheers
Paul

**Comment by JamesNK** [comment]
Glad you've fixed your issue. Closing the issue as resolved.
